doc/bitcoin-conf.md

```diff
@@ -30,6 +30,19 @@ Network specific options can be:
 - placed into sections with headers `[main]` (not `[mainnet]`), `[test]` (not `[testnet]`) or `[regtest]`;
 - prefixed with a chain name; e.g., `regtest.maxmempool=100`.
 
+Options prefixed with a chain name take precedence over options placed into sections with headers, which themselves
```

Comment:
I think this is only true because there's no syntax for ending sections in the config file. Options prefixed with a section name and options written inside a section aren't treated differently or given different precedence. The reason why the rpcport is 3000 instead of 4000, is that the 3000 line is above the 4000 line and when options are duplicated in the config file, the first value is usually the one that sticks.

Reply:
That sounds correct to me. IMO it'd be a good idea to have syntax to end sections, but I'd say that's outside the scope of this PR.

Reply:
https://github.com/bitcoin/bitcoin/blob/76e2cded477bc483ec610212bdadf21fe35292d4/src/util/system.cpp#L229-L237

Reply:
`ini` format doesn't have a way to end sections, and IMO also doesn't particularly need it

Reply:
My comment is saying the new documentation is wrong. It is not saying that that format should have a way to end sections. It is not true that "Options prefixed with a chain name take precedence over options placed into sections with headers." In fact, `section.option=value` and `[section]↵option=value` are interpreted exactly the same and have the same precedence. The difference in the example just comes from the order values are specified in the config file.

If we were going to change the format, my priority would be to first clean up the inconsistent behaviors documented in #15934: (command line [ignored arguments](https://github.com/ryanofsky/bitcoin/blob/1d543adae593bdbfd954e80ed61ac907db0c5a7b/src/util/system.cpp#L326-L330) and [more ignored arguments](https://github.com/ryanofsky/bitcoin/blob/1d543adae593bdbfd954e80ed61ac907db0c5a7b/src/util/system.cpp#L186-L191), and config file [reverse precedence](https://github.com/ryanofsky/bitcoin/blob/1d543adae593bdbfd954e80ed61ac907db0c5a7b/src/util/settings.cpp#L63-L67), [inconsistently applied top-level settings](https://github.com/ryanofsky/bitcoin/blob/1d543adae593bdbfd954e80ed61ac907db0c5a7b/src/util/settings.cpp#L57-L61), and [zombie values](https://github.com/ryanofsky/bitcoin/blob/1d543adae593bdbfd954e80ed61ac907db0c5a7b/src/util/settings.cpp#L100-L107)).

Reply:
> ... and zombie values

IIIC, these zombie values appeared after 4d34fcc7138f0ffc831f0f8601c50cc7f494c197.

Reply:
@ryanofsky Do you have an example config file where what I described in this PR doesn't hold?